wire-grpc-client/src/main/java/com/squareup/wire/GrpcClient.kt

```diff
@@ -50,17 +53,27 @@ class GrpcClient private constructor(
     return Proxy.newProxyInstance(service.java.classLoader, arrayOf<Class<*>>(service.java),
         object : InvocationHandler {
           override fun invoke(proxy: Any, method: Method, args: Array<out Any>): Any? {
-            val context = args[0] as CoroutineContext
+            val continuation = args.last() as Continuation<Any>
 
             return when (val grpcMethod = methodToService[method] as GrpcMethod<*, *>) {
-              is GrpcMethod.RequestResponse ->
-                context.invokeSuspending(args[2] as Continuation<Any>) {
-                  grpcMethod.invoke(context, this@GrpcClient, parameter = args[1])
+              is GrpcMethod.RequestResponse -> {
+                continuation.invokeSuspending {
+                  grpcMethod.invoke(continuation.context, this@GrpcClient, parameter = args[0])
                 }
-              is GrpcMethod.StreamingResponse ->
-                grpcMethod.invoke(context, this@GrpcClient, parameter = args[1])
-              is GrpcMethod.StreamingRequest -> grpcMethod.invoke(context, this@GrpcClient)
-              is GrpcMethod.FullDuplex -> grpcMethod.invoke(context, this@GrpcClient)
+              }
+              is GrpcMethod.StreamingResponse -> {
+                continuation.resume(
```

Comment:
Doing this is forbidden. If you have the result available synchronously you need to just return it directly.

It's also a strong indicator that this shouldn't be a suspending function because it never suspends.